### ISSUE TITLE
Unable to configure 1.5.x, Unexpected exception

### ISSUE THREAD

**Issue by kimfr** [hawky358/hass_ids_hyyp#22]
Hello,

When I try and add the integration, once I have filled out username, password and selected "ADT Securehome" - I immediately get an "Unexpected error".

I can see the following in the HA log:
2023-11-04 11:23:29.685 ERROR (MainThread) [custom_components.ids_hyyp.config_flow] Unexpected exception
Traceback (most recent call last):
  File "/config/custom_components/ids_hyyp/config_flow.py", line 96, in async_step_user
    token_data = await self.hass.async_add_executor_job(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ids_hyyp/config_flow.py", line 48, in _validate_and_create_auth
    _initialize_push_receiver()
  File "/config/custom_components/ids_hyyp/config_flow.py", line 40, in _initialize_push_receiver
    fcmfile = open(FCM_PERSISTENTIDFILE, "w")
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/share/ids_hyyp_persistent_ids.json'

This is the first time I trying to use your integration. I tried 1.5.1 - also fails with the same message - 1.4.1 however works fine.

I'm running HA 2023.10.5, Docker container on a RPi.

**Comment by hawky358** [comment]
The issue is likely due to you running the container version of HA.
The access to system resources is probably a bit different.

I can't reproduce the error as such since I don't have a setup like that.

I'll see if I can develop an alternative method where it will not be required to save persistent data for these notification so this will not be relevant anymore.

I don't know if there are other functions as well which will work differently on the container version.

**Comment by kimfr** [comment]
> The access to system resources is probably a bit different.
Do you perhaps have a list of system resources you are using?
> 
> I can't reproduce the error as such since I don't have a setup like that.
Happy to help with testing.
>
> I'll see if I can develop an alternative method where it will not be required to save persistent data for these notification so this will not be relevant anymore.
>
From a container perspective, persistent data is mapped into the container as /config - /etc/localtime is also mapped in together with /run/dbus (needed for Bluetooth).

**Comment by hawky358** [comment]
Hello,

I believe some indirectly related changes I did a few months ago also negated the need for this persistent storage across reboots, but I never tested it since that wasn't the focus.

As a starting point for your issue, I've created a test branch that you can try. I want to see whether you can get going before I push it for wider testing as a release.

Will you be able to manually load this onto your HASS, replacing the current version? 
Note if you are using 1.4.x you must remove the integration and add from "scratch" to load some newer stored data.

[https://github.com/hawky358/hass_ids_hyyp/tree/push-receiver-storage-test-fix](https://github.com/hawky358/hass_ids_hyyp/tree/push-receiver-storage-test-fix)

**Comment by hawky358** [comment]
Hey,

I didn't mean you should update the repository via HACS, I meant you use SSH to transfer the integration into custom_components (i.e. "pre-HACS") method.

If this is not possible, I'll have to create a release .
Let me know.

**Comment by kimfr** [comment]
Hey,

No problem ;-) - hopefully I did the correctly now. I downloaded the zip, unzip and then copied the custom_components/ids_hyyp/ content directly into the HA HACS custom_components/ids_hyyp/ directory.

I was able to add the Integration now and successfully complete the configuration/registration.

Now all I need to do is understand exactly how to use it, I'm looking for:
- Multiple stay profiles and go through the Cards Code discussion
- Get Push notification to work
